compiler/src/dotty/tools/dotc/core/PatternTypeConstrainer.scala

```diff
@@ -189,16 +189,14 @@ trait PatternTypeConstrainer { self: TypeComparer =>
       case _ => false
     }
 
-    def widenVariantParams = new TypeMap {
-      def apply(tp: Type) = mapOver(tp) match {
-        case tp @ AppliedType(tycon, args) =>
-          val args1 = args.zipWithConserve(tycon.typeParams)((arg, tparam) =>
-            if (tparam.paramVarianceSign != 0) TypeBounds.empty else arg
-          )
-          tp.derivedAppliedType(tycon, args1)
-        case tp =>
-          tp
-      }
+    def widenVariantParams(tp: Type) = tp match {
```

Comment:
I don't quite understand what all this stuff is doing, but are we intentionally not dealiasing here?

Reply:
The type is already dealiased in `constrainPatternType`, we need to dealias in order to correctly handle intersections, unions and other things.